go/vt/mysqlctl/query.go

```diff
@@ -71,13 +71,54 @@ func (mysqld *Mysqld) ExecuteSuperQueryList(ctx context.Context, queryList []str
 func (mysqld *Mysqld) executeSuperQueryListConn(ctx context.Context, conn *dbconnpool.PooledDBConnection, queryList []string) error {
 	for _, query := range queryList {
 		log.Infof("exec %v", redactMasterPassword(query))
-		if _, err := mysqld.executeFetchContext(ctx, conn, query, 10000, false); err != nil {
+		_, err := mysqld.executeFetchContext(ctx, conn, query, 10000, false)
+		switch {
+		case err == nil:
+			return nil
+
+		case strings.ToUpper(query) == "START SLAVE" && isReplicationErr1872(err):
```

Comment:
I don't think a generic "execute query" function is the right place for this specific logic. If there's not a subroutine like `mysqld.StartSlave()` that all the code paths you're trying to catch use, then we should add one and change the appropriate sites to use it.

Or alternatively, would it work for you if we only check for this error in the repair section of the replication health reporter? That seems safer than injecting the fix at this low level, unless for some reason the periodic repair doesn't work for your needs.

Reply:
Yeah, it felt weird to do this at this low level. I started at `repairReplication` and kept digging. That code path never called `StartSlave` that I found.

Reply:
If you're ok with only doing this check in the `repairReplication` code path, then we can just make sure sufficient error detail gets propagated back up to that level to detect this specific case, and put the logic there. WDYT?

Reply:
That works for me. This was just a quick stab in the dark to get some feedback.